relayer/ibc-client.go

```diff
@@ -0,0 +1,130 @@
+package relayer
+
+import (
+	"fmt"
+
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/core/02-client/types"
+	ibcexported "github.com/cosmos/cosmos-sdk/x/ibc/core/exported"
+	tmclient "github.com/cosmos/cosmos-sdk/x/ibc/light-clients/07-tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+	"golang.org/x/sync/errgroup"
+)
+
+// GetIBCCreateClientHeader updates the off chain tendermint light client and
+// returns an IBC Update Header which can be used to create an on-chain
+// light client.
+func (c *Chain) GetIBCCreateClientHeader(dst *Chain) (*tmclient.Header, error) {
+	lightBlock, err := c.UpdateLightClient()
+	if err != nil {
+		return nil, err
+	}
+
+	protoVal, err := tmtypes.NewValidatorSet(lightBlock.ValidatorSet.Validators).ToProto()
+	if err != nil {
+		return nil, err
+	}
+
+	return &tmclient.Header{
+		SignedHeader: lightBlock.SignedHeader.ToProto(),
+		ValidatorSet: protoVal,
+	}, nil
+}
+
+// GetIBCCreateClientHeaders returns the IBC TM header which will create an on-chain
+// light client. The headers do not have trusted headers (ie trusted validators
+// and trusted height)
+func GetIBCCreateClientHeaders(src, dst *Chain) (srcHeader, dstHeader *tmclient.Header, err error) {
+	var eg = new(errgroup.Group)
+	eg.Go(func() error {
+		srcHeader, err = src.GetIBCCreateClientHeader(dst)
+		return err
+	})
+	eg.Go(func() error {
+		dstHeader, err = dst.GetIBCCreateClientHeader(src)
+		return err
+	})
+	if err = eg.Wait(); err != nil {
+		return
+	}
+	return
+
+}
+
+// GetIBCUpdateHeader updates the off chain tendermint light client and
+// returns an IBC Update Header which can be used to update an on chain
+// light client.
+func (c *Chain) GetIBCUpdateHeader(dst *Chain) (*tmclient.Header, error) {
+	h, err := c.GetIBCCreateClientHeader(dst)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.InjectTrustedFields(dst, h)
+}
+
+// GetIBCUpdateHeaders return the IBC TM Header which will update an on-chain
+// light client. A header for the source and destination chain is returned.
+func GetIBCUpdateHeaders(src, dst *Chain) (srcHeader, dstHeader *tmclient.Header, err error) {
+	var eg = new(errgroup.Group)
+	eg.Go(func() error {
+		srcHeader, err = src.GetIBCUpdateHeader(dst)
+		return err
+	})
+	eg.Go(func() error {
+		dstHeader, err = dst.GetIBCUpdateHeader(src)
+		return err
+	})
+	if err = eg.Wait(); err != nil {
+		return
+	}
+	return
+
+}
+
+// InjectTrustedFields injects the necessary trusted fields for a header to update a light
+// client stored on the destination chain, using the information provided by the source
+// chain.
+// TrustedHeight is the latest height of the IBC client on dst
+// TrustedValidators is the validator set of srcChain at the TrustedHeight
+// InjectTrustedFields returns a copy of the header with TrustedFields modified
+func (c *Chain) InjectTrustedFields(dst *Chain, header *tmclient.Header) (*tmclient.Header, error) {
```

Comment:
ditto

Reply:
not necessary. Trusted fields is a tendermint specific concept